copulas/__init__.py

```diff
@@ -21,19 +21,29 @@ class NotFittedError(Exception):
 
 
 @contextlib.contextmanager
-def random_seed(seed):
-    """Context manager for managing the random seed.
+def set_random_state(random_state, set_model_random_state):
+    """Context manager for managing the random state.
 
     Args:
-        seed (int):
-            The random seed.
+        random_state (int or np.random.RandomState):
+            The random seed or RandomState.
+        set_model_random_state (function):
+            Function to set the random state on the model.
     """
-    state = np.random.get_state()
-    np.random.seed(seed)
+    original_state = np.random.get_state()
+
+    if isinstance(random_state, int):
```

Comment:
This `if` block should have 3 branches: `int`, `RandomState` and `tuple`, since passing a state like the one returned by `np.random.get_state()` should also be possible (see comment below)

```python3
if `int`:
    state = RandomState(state).get_state()
elif `RandomState`:
    state = state.get_state()
elif not `tuple`:
    raise TypeError()
```